web/oauth/claims.py

```diff
@@ -0,0 +1,37 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def process(userinfo: dict, expected_claims: list[str]) -> tuple[list[str], dict[str, str]]:
+    """Process expected claims from the userinfo dict.
+
+    - Boolean claims comes back in userinfo like `{ "claim": "1" | "0" }` or `{ "claim": "true" }`
+    - Other claims come back in userinfo like `{ "claim": "value" }`
+
+    Returns a tuple `(claims: list[str], errors: dict[str, int])`
+    """
+    claims = []
+    errors = {}
+
+    for claim in expected_claims:
+        claim_value = userinfo.get(claim)
+        if not claim_value:
+            logger.warning(f"userinfo did not contain: {claim}")
+        try:
+            claim_value = int(claim_value)
+        except (TypeError, ValueError):
+            pass
+        if isinstance(claim_value, int):
+            if claim_value == 1:
+                # if userinfo contains our claim and the flag is 1 (true), store the *claim*
+                claims.append(claim)
+            elif claim_value >= 10:
+                errors[claim] = claim_value
+        elif isinstance(claim_value, str):
+            if claim_value.lower() == "true":
+                claims.append(claim)
+            elif claim_value.lower() != "false":
+                claims.append(f"{claim}:{claim_value}")
```

Comment:
Just a thought: if we're doing this for `"false"`, maybe we should also do it for `0`?

This doesn't need to be changed for this PR.

Reply:
_(I don't know if this is 100% right because the `true/false` claims are new for us -- cc @johnatstate)_

Here's my thinking on the integer cases:

* Store the name of the claim if `value == 1` (success)
* Store the name and the value if `value > 10` (error)
* We don't store anything if `value == 0` (fail)

And so this is analogous for the non-integer cases:

* Store the name of the claim if `value == true` (success)
* Store the name and the value if `value != false` (an actual value, like `email@domain.com`)
* We don't store anything if `value == false` (fail)

That last bullet is an assumption -- I don't know if we would get `false` back as a string, but I saw `true` so added the corollary.

Reply:
Oh I see. Thanks for the clarification!